builtin/providers/aws/resource_aws_emr.go

```diff
@@ -0,0 +1,439 @@
+			Ec2SubnetId:                 aws.String(subnet),
+			InstanceCount:               aws.Int64(int64(coreInstanceCount + 1)),
+			KeepJobFlowAliveWhenNoSteps: aws.Bool(true),
+			MasterInstanceType:          aws.String(masterInstanceType),
+			SlaveInstanceType:           aws.String(coreInstanceType),
+			TerminationProtected:        aws.Bool(false),
+			AdditionalMasterSecurityGroups: []*string{
+				aws.String(extraMasterSecGrp),
+			},
+			AdditionalSlaveSecurityGroups: []*string{
+				aws.String(extraSlaveSecGrp),
+			},
+			EmrManagedMasterSecurityGroup: aws.String(emrMasterSecGrp),
+			EmrManagedSlaveSecurityGroup:  aws.String(emrSlaveSecGrp),
+		},
+		Name:         aws.String(d.Get("name").(string)),
+		Applications: emrApps,
+
+		JobFlowRole:       aws.String(instanceProfile),
+		ReleaseLabel:      aws.String(d.Get("release_label").(string)),
+		ServiceRole:       aws.String(serviceRole),
+		VisibleToAllUsers: aws.Bool(d.Get("visible_to_all_users").(bool)),
+	}
+
+	if v, ok := d.GetOk("log_uri"); ok {
+		logUrl := v.(string)
+		params.LogUri = aws.String(logUrl)
+	}
+	if v, ok := d.GetOk("bootstrap_action"); ok {
+		bootstrapActions := v.(*schema.Set).List()
+		log.Printf("[DEBUG] %v\n", bootstrapActions)
+		params.BootstrapActions = expandBootstrapActions(bootstrapActions)
+	}
+	if v, ok := d.GetOk("tags"); ok {
+		tagsIn := v.(map[string]interface{})
+		params.Tags = expandTags(tagsIn)
+	}
+	if v, ok := d.GetOk("configurations"); ok {
+		confUrl := v.(string)
+		params.Configurations = expandConfigures(confUrl)
+	}
+
+	log.Printf("[DEBUG] EMR Cluster create options: %s", params)
+	resp, err := conn.RunJobFlow(params)
+
+	if err != nil {
+		log.Printf("[ERROR] %s", err)
+		return err
+	}
+
+	log.Printf("[DEBUG] Created EMR Cluster done...")
+	d.SetId(*resp.JobFlowId)
+
+	return resourceAwsEMRRead(d, meta)
+}
+
+func resourceAwsEMRRead(d *schema.ResourceData, meta interface{}) error {
+	emrconn := meta.(*AWSClient).emrconn
+
+	req := &emr.DescribeClusterInput{
+		ClusterId: aws.String(d.Id()),
+	}
+
+	resp, err := emrconn.DescribeCluster(req)
+	if err != nil {
+		return fmt.Errorf("Error reading EMR cluster: %s", err)
+	}
+
+	if resp.Cluster == nil {
+		d.SetId("")
+		log.Printf("[DEBUG] EMR Cluster (%s) not found", d.Id())
+		return nil
+	}
+
+	instance := resp.Cluster
+
+	if instance.Status != nil {
+		if *resp.Cluster.Status.State == "TERMINATED" {
+			d.SetId("")
+			log.Printf("[DEBUG] EMR Cluster (%s) was TERMINATED already", d.Id())
+			return nil
+		}
+
+		if *resp.Cluster.Status.State == "TERMINATED_WITH_ERRORS" {
+			d.SetId("")
+			log.Printf("[DEBUG] EMR Cluster (%s) was TERMINATED_WITH_ERRORS already", d.Id())
+			return nil
+		}
+	}
+
+	instanceGroups, errGrps := loadGroups(d, meta)
+	if errGrps == nil {
+		coreGroup := findGroup(instanceGroups, "CORE")
+		d.Set("core_instance_type", coreGroup.InstanceType)
+	}
+
+	return nil
+}
+
+func resourceAwsEMRUpdate(d *schema.ResourceData, meta interface{}) error {
```

Comment:
- [x] add `d.HasChange("core_instance_count") ` checking

You are right
`core_instance_count` is the only attribute can be changed